### DISCUSSION TITLE
Getting Started, MLFlow logger.

### DISCUSSION THREAD

**Discussion by MoAbbasid** [weecology/DeepForest#677] [General]
Hello, I have a few questions about DeepForest
Im wondering if this deepforest model is still the best to use for tree counting, or have other models like Ultralytics YOLO or the VLMs do better?
also do annotations have to be in format "image_path, xmin,ymin,xmax,ymax, class" ? 
also I would like an example of how to run this and have it log the metrics and params with Mlflow ?
And thank you

**Comment by bw4sz** [comment]
Howdy @MoAbbasid , 

The backbone model is really meant for fine-tuning based on local annotations. Yes, it works, and whether its the 'best' is really hard to say, but its super-useful and that's enough for me. Data always matters more than architecture. You can annotate as a shapefile and use the shapefile_to_annotations function. 
https://deepforest.readthedocs.io/en/latest/source/deepforest.html#deepforest.utilities.shapefile_to_annotations. 

I haven't used Mlflow, but a quick google check suggests there is a pytorch lightning logger

https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.loggers.mlflow.html

So as a guess (did not run it), something like

```
from lightning.pytorch import Trainer
from lightning.pytorch.loggers import MLFlowLogger
from deepforest import main

m = main.deepforest()
m.use_release()

m.config["train"]["csv_file"] 
m.config["train"]["root_dir"]

mlf_logger = MLFlowLogger(experiment_name="lightning_logs", tracking_uri="file:./ml-runs")
m.create_trainer(logger=mlf_logger)
m.trainer.fit(m)
```

Report back and we can add to docs to others.

**Reply by MoAbbasid** [reply]
Hi, @bw4sz, 
Thanks for your response, 
I will check the Mlflow implementation and report back on it,
also I saw that the Deepforest docs about shapefile annotation, and the docs mention label-studio for annotations, I use a tool similar to it that outputs bounding box annotations in json format, how do I load these into the model for finetuning? I havent been able to locate an example in the docs

**Reply by bw4sz** [reply]
Can you show an example box and we can make a pull request to help other users.

**Reply by MoAbbasid** [reply]
this tool is called Anylabeling for example, the bboxes are automatically saved next to the image in json format like this:

**Comment by bw4sz** [comment]
drop a json in here, not the image, and we can write a parser.

On Mon, Jun 10, 2024 at 7:54 PM Abbas Jr ***@***.***> wrote:

> image.png (view on web)
> <https://github.com/weecology/DeepForest/assets/109457597/e43d742d-4f3a-4c52-a288-8c40708225a5>
> this tool is called Anylabeling for example, the bboxes are automatically
> saved next to the image in json format like this:
> image.png (view on web)
> <https://github.com/weecology/DeepForest/assets/109457597/937d30e8-e2a0-4e6a-8108-8c7e9c673e35>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/weecology/DeepForest/discussions/677#discussioncomment-9733591>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJHBLBUG56DSE4SEIQJGE3ZGZRHBAVCNFSM6AAAAABI33CIMOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMZTGU4TC>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Ben Weinstein, Ph.D.
Research Scientist
University of Florida

**Reply by MoAbbasid** [reply]
hi, this is an example of a sample json output from the annotator:
```
{
  "version": "2.3.6",
  "flags": {},
  "shapes": [
    {
      "label": "Tree",
      "points": [
        [
          810.3333333333335,
          601.3888888888889
        ],
        [
          1021.4444444444446,
          601.3888888888889
        ],
        [
          1021.4444444444446,
          762.5
        ],
        [
          810.3333333333335,
          762.5
        ]
      ],
      "group_id": null,
      "description": "",
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {}
    },
    {
      "label": "Tree",
      "points": [
        [
          1099.2222222222224,
          851.3888888888889
        ],
        [
          1271.4444444444446,
          851.3888888888889
        ],
        [
          1271.4444444444446,
          1006.9444444444445
        ],
        [
          1099.2222222222224,
          1006.9444444444445
        ]
      ],
      "group_id": null,
      "description": "",
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {}
    },
    {
      "label": "Tree",
      "points": [
        [
          2165.8888888888887,
          829.1666666666664
        ],
        [
          2377.0,
          829.1666666666664
        ],
        [
          2377.0,
          1062.5000000000002
        ],
        [
          2165.8888888888887,
          1062.5000000000002
        ]
      ],
      "group_id": null,
      "description": "",
      "difficult": false,
      "shape_type": "rectangle",
      "flags": {},
      "attributes": {}
    }
  ],
  "imagePath": "1.jpg",
  "imageData": null,
  "imageHeight": 4825,
  "imageWidth": 6754,
  "text": ""
}
```
